crates/hir/src/hir_def/mod.rs

```diff
@@ -80,11 +83,18 @@ impl<'db> HirIngot<'db> for Ingot<'db> {
     fn resolved_external_ingots(self, db: &'db dyn HirDb) -> Vec<(IdentId<'db>, Ingot<'db>)> {
         self.dependencies(db)
             .iter()
-            .filter_map(|(name, url)| {
-                self.index(db)
-                    .containing_ingot(db, url)
-                    .map(|ingot_description| (IdentId::new(db, name), ingot_description))
-            })
+            .filter_map(
+                |Dependency {
+                     alias,
+                     description: DependencyDescription { url, .. },
```

Comment:
The oversimplified `DependencyDescription` with its handy `Url` is making this a little too easy.  In the end we need to be able to consistently map a `Dependency` to a URL.  In the case of relative path dependencies, there needs to be some way of injecting a base URL so that if the dependency points to a relative path we can resolve an absolute URL for that relative path.

Maybe we can have a method like:
```rust
impl Dependency {
  fn url(&self, base_url: Url) -> Url {
    let flattened_absolute_url = match self.location ... // handle path vs url
    return flattened_absolute_url
}
```

The idea would be that the dependency itself carries around either a url or a path and that this can be resolved given a base url.